modules/internal/ChapelBase.chpl

```diff
@@ -912,12 +912,12 @@ module ChapelBase {
     }
   }
 
-  pragma "unsafe" // work around problems storing non-nilable classes
   proc init_elts(x, s, type t) : void {
     var initMethod = chpl_getArrayInitMethod();
 
     // for uints, check that s > 0, so the `s-1` below doesn't overflow
-    if isUint(s) && s == 0 {
+    // does not need to init for ints, either
```

Comment:
I think a better comment here would say something like "No need to init an array with zero elements"